src/System.Diagnostics.TraceSource/tests/TraceEventCacheClassTests.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+
+namespace System.Diagnostics.TraceSourceTests
+{
+    public class TraceEventCacheClassTests
+    {
+        [Fact]
+        public void DateTimePropertyTest()
+        {
+            var cache = new TraceEventCache();
+            var begin = DateTime.UtcNow;
+            var dt = cache.DateTime;
+            var end = DateTime.UtcNow;
+            Assert.True(dt >= begin);
+            Assert.True(dt <= end);
+            var cached = cache.DateTime;
+            Assert.Equal(dt, cached);
+        }
+
+        [Fact]
+        public void ProcessIdTest()
+        {
+            var cache = new TraceEventCache();
+            var id = cache.ProcessId;
+            // TODO: This was causing a compile error on the linux CI build. 
+            // (I don't have an environment to replicate this in)
+            //var expected = Interop.mincore.GetCurrentProcessId();
```

Comment:
You do, actually :smile: Just pass /p:OS=Linux to msbuild when building, e.g. in your tests directory do "msbuild /t:rebuild /t:BuildAndTest /p:OS=Linux".

The problem is that GetCurrentProcessId is a P/Invoke to a Win32 function, and it's defined in a file that's only included when building for Windows, i.e. /p:OS=Windows_NT.  If you want to validate that cache.ProcessId is the ID of the current process, you can update the packages.config for the test assembly to include System.Diagnostics.Process, and then here you can use Process.GetCurrentProcess().Id.  That'll be available and work on Linux.